src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/BackupOperation/BackupOperation.cs

```diff
@@ -149,6 +150,28 @@ public BackupConfigInfo CreateBackupConfigInfo(string databaseName)
         }
 
         /// <summary>
+        /// Generate script for backup
+        /// </summary>
+        public string ScriptBackup()
+        {
+            StringBuilder sb = new StringBuilder();            
+            SqlExecutionModes oldExecutionMode = this.dataContainer.Server.ConnectionContext.SqlExecutionModes;
+            this.dataContainer.Server.ConnectionContext.SqlExecutionModes = SqlExecutionModes.CaptureSql;
+            this.dataContainer.Server.ConnectionContext.CapturedSql.Clear();
+
+            this.PerformBackup();
+
+            foreach (String s in this.dataContainer.Server.ConnectionContext.CapturedSql.Text)
+            {
+                sb.AppendLine(s);
```

Comment:
I think this will add \n as end of line, you should do append and add the Environment End of  line. or test the result in mac